quickwit/quickwit-control-plane/src/control_plane.rs

```diff
@@ -155,6 +162,36 @@ impl Handler<ControlPlanLoop> for ControlPlane {
     }
 }
 
+fn convert_metastore_error<T>(
+    metastore_error: MetastoreError,
+) -> Result<ControlPlaneResult<T>, ActorExitStatus> {
+    // If true, we know that the transactions has not been recorded in the Metastore.
+    // If false, we simply are not sure whether the transaction has been recorded or not.
+    let metastore_failure_is_certain = match &metastore_error {
+        MetastoreError::AlreadyExists(_)
+        | MetastoreError::FailedPrecondition { .. }
+        | MetastoreError::Forbidden { .. }
+        | MetastoreError::InvalidArgument { .. }
+        | MetastoreError::JsonDeserializeError { .. }
+        | MetastoreError::JsonSerializeError { .. }
+        | MetastoreError::NotFound(_) => true,
+        MetastoreError::Unavailable(_)
+        | MetastoreError::Internal { .. }
+        | MetastoreError::Io { .. }
+        | MetastoreError::Connection { .. }
+        | MetastoreError::Db { .. } => false,
+    };
+    if metastore_failure_is_certain {
+        // If the metastore failure is certain, this is actually a good thing.
+        // We do not need to restart the control plane.
+        Ok(Err(ControlPlaneError::Metastore(metastore_error)))
+    } else {
+        // If the metastore failure is uncertain, we need to restart the control plane
+        // so that it gets resynced with the metastore state.
+        Err(ActorExitStatus::from(anyhow::anyhow!(metastore_error)))
```

Comment:
Do we make sure we're correctly logging an error when that happens? Additionally, we should emit a metric when a control plane crash/restart occurs. As an operator, I'd certainly want to be able to set an alert on that event.

Reply:
nice catch! This is a very good point.

Reply:
ill add the log in this PR and add the counter in a different PR. I created an issue to avoid forgetting